### DISCUSSION TITLE
How about putting a header file that consolidates configuration definitions under the `programs/` directory?

### DISCUSSION THREAD

**Discussion by dearblue** [lz4/lz4#1344] [Ideas]
For example, `programs/lz4conf.h`, `programs/lz4def.h`, etc.?

I think it would also be useful as an alternative to documentation.

**Comment by t-mat** [comment]
I suppose we can also think about similar "configuration header" as `lib/lz4_conf.h`, `lib/lz4frame_conf.h`, etc.

**Comment by Cyan4973** [comment]
> I suppose we can also think about similar "configuration header" as `lib/lz4_conf.h`, `lib/lz4frame_conf.h`, etc.

`lib/` has a different philosophy:
it must also reduce the nb of files it exposes,
in order to make it easy to integrate in 3rd party projects
by just copy/pasting the necessary files into the target project,
with the lowest level being just `lz4.c` + `lz4.h`.

Hence, I prefer avoiding multi-level header chains of dependency there.

No such constraint in `programs/`, 
it's only designed to be portable.

**Comment by dearblue** [comment]
> `lib/` has a different philosophy:

I understand and agree with that.
And with the merge of #1343, I feel that this discussion is in the closing phase.
